### ISSUE TITLE
The debugger crashes randomly during debugging.

### ISSUE THREAD

**Issue by Lingondary** [godotengine/godot#90799]
### Tested versions

4.2.1stable.mono

### System information

Windows 10, Godot 4.2.1stable.mono

### Issue description

During debugging, at a random stage of debugging, the game closes without any errors, and the debugging process stops. There are no errors following this. Without debugging, the game runs normally and does not close during operation.
Within the limits of my knowledge and skills, I looked through all the code where the error started to appear and also tried to detect it with the help of logging. But since the error occurs at a random moment of working with the inventory, I could not trace it.
Video: https://www.youtube.com/watch?v=9m1fz-Kz0w4&ab_channel=Lingondary

### Steps to reproduce

https://github.com/Lingondary/ARPG_Godot/blob/main/Inventory/Scripts/InventoryUI.gd - In _on_inventory_updated(): the debugger is more likely to crash. It happens both before and after its execution.

### Minimal reproduction project (MRP)

https://github.com/Lingondary/ARPG_Godot/tree/main

**Comment by AThousandShips** [comment]
The link to your repo is broken, also this doesn't look like a minimal project just from the title, so it'd be better if you provide a minimal project for testing

**Comment by Lingondary** [comment]
> The link to your repo is broken, also this doesn't look like a minimal project just from the title, so it'd be better if you provide a minimal project for testing

I apologize, I made the repository public.
Unfortunately, I can't make the minimal reproduction project really minimal, as I don't understand the nature of the error occurrence. 😢

**Comment by Lingondary** [comment]
> That means you should get and replicate it and create a minimal project by testing things, if it's hard for you to replicate it like that it'll be harder for everyone else

Sorry, I'll leave it for information, the problem is reproduced only on my PC in any project using debug in GDScirpt.
I have checked the work of my project on 2 other PCs with Win 10.

Configurations:
3080 is running on both of them
Processors: 
Ryzen 7 7800x3d
Ryzen 7 5800X
No problems on both systems.

On mine, I have a GTX 1080Ti and Ryzen5 3500 installed.
In addition to testing on my working Win 10, I also tried running the project on a clean virtual Win10 and on Ubutnu 22.04, the problem also occurred in both cases.

**Comment by LowFire** [comment]
I have this exact issue on my main game project. Sometimes when I hit a break point or there is an error in the code, the game closes immediately and the debugging process stops. I have no idea why it happens and haven't been able to pinpoint why. Usually when the debugger starts crashing randomly, a computer restart seems to fix it. I don't have this problem on any of my other projects, just the main game project I'm working on, so it might be project specific.

**Comment by LowFire** [comment]
I decided to run my game project in a dev build of the engine and got a crash dump of when the debugger crashes. I have no idea if this is helpful or not, but hopefully it provides some insight? I have no idea what any of it means.

![DebuggerCrash](https://github.com/user-attachments/assets/ae0bb364-df02-4780-ad72-038bfd59f2e5)

**Comment by Calinou** [comment]
According to the backtrace you posted, the crash occurs somewhere around here: https://github.com/godotengine/godot/blob/4bef4d9808848c38c3f285edb300b4f3a843e543/modules/gdscript/gdscript_editor.cpp#L404-L411

**Comment by LowFire** [comment]
> According to the backtrace you posted, the crash occurs somewhere around here:
> 
> https://github.com/godotengine/godot/blob/4bef4d9808848c38c3f285edb300b4f3a843e543/modules/gdscript/gdscript_editor.cpp#L404-L411

Ok interesting. So it's probably related to this issue then https://github.com/godotengine/godot/issues/92607#issue-2328603836. The OP determined that this exact code is what was causing the debugger to crash.

Does this mean that this bug is already fixed in the upcoming version of godot? The issue had been closed not too long ago.

**Comment by akien-mga** [comment]
> Does this mean that this bug is already fixed in the upcoming version of godot? The issue had been closed not too long ago.

It might well be fixed already, yes. If you can test the just released 4.3.rc3, you might be able to confirm that: https://godotengine.org/article/release-candidate-godot-4-3-rc-3/

**Comment by LowFire** [comment]
> It might well be fixed already, yes. If you can test the just released 4.3.rc3, you might be able to confirm that: https://godotengine.org/article/release-candidate-godot-4-3-rc-3/

Yeah, it seems to have been fixed. The debugger is not crashing when I'm using the release candidate version. Thank goodness.

**Comment by Calinou** [comment]
@Lingondary Can you test 4.3.rc3 on your project and see if the issue is resolved?
